# FORTUNE 500 COMPANIES - 2020 : Insights from Data

The Fortune 500 is an annual list compiled and published by Fortune magazine that ranks 500 of the largest United States corporations by the annual total revenue using publicly available data.

The Fortune 500, created by Edgar P. Smith, was first published in 1955.





In [1]:
import numpy as np 
import pandas as pd 

In [4]:
df = pd.read_csv('Fortune 2020.csv')

## Top 10 Fortune 500 Companies

In [6]:
df.head(10)

rank             company no_of_employees rank_change    revenues  \
0     1             Walmart    2,200,000.00           -  523,964.00   
1     2          Amazon.com      798,000.00        3.00  280,522.00   
2     3         Exxon Mobil       74,900.00       -1.00  264,938.00   
3     4               Apple      137,000.00       -1.00  260,174.00   
4     5          CVS Health      290,000.00        3.00  256,776.00   
5     6  Berkshire Hathaway      391,500.00       -2.00  254,616.00   
6     7  UnitedHealth Group      325,000.00       -1.00  242,155.00   
7     8            McKesson       70,000.00       -1.00  214,319.00   
8     9                AT&T      247,800.00           -  181,193.00   
9    10   AmerisourceBergen       21,500.00           -  179,589.10   

  revenue_change    profits profit_change      assets  market_value  
0           0.02  14,881.00          1.23  236,495.00    321,803.30  
1           0.21  11,588.00          0.15  225,248.00    970,680.10  
2          -0.09  14,340.00         -0.31  362,597.00    160,696.30  
3          -0.02  55,256.00         -0.07  338,516.00  1,112,640.80  
4           0.32   6,634.00             -  222,449.00     77,375.80  
5           0.03  81,417.00         19.25  817,729.00    442,897.10  
6           0.07  13,839.00          0.16  173,889.00    236,555.20  
7           0.03      34.00         -0.49   59,672.00     21,845.00  
8           0.06  13,903.00         -0.28  551,669.00    209,387.50  
9           0.07     855.40         -0.48   39,172.00     18,221.40

In [7]:
df.shape

(500, 10)


There are 500 Companies in this table with 10 Columns in which 

> 1st column indicates Rank , 

> 2nd column is dedicated to the Company name and 

> The rest 8 columns provide information about each company


In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   rank             500 non-null    int64 
 1   company          500 non-null    object
 2   no_of_employees  500 non-null    object
 3   rank_change      500 non-null    object
 4   revenues         500 non-null    object
 5   revenue_change   500 non-null    object
 6   profits          500 non-null    object
 7   profit_change    500 non-null    object
 8   assets           500 non-null    object
 9   market_value     500 non-null    object
dtypes: int64(1), object(9)
memory usage: 39.2+ KB


### Datatype Conversion : From object to float


In [13]:
df = df.replace('-','0')

In [14]:
def remove_comma(x):
  x = x.replace(',','')
  return x

In [15]:
cols_comma_remove = ['no_of_employees','revenues','profits','assets','market_value' ]
for i in cols_comma_remove:
  df[i] = df[i].apply(remove_comma)

In [16]:
for i in df.columns[2:]:
  df[i] = df[i].astype(float)
for i in df.columns:
  print(i, ' : ', df[i].dtype)

rank  :  int64
company  :  object
no_of_employees  :  float64
rank_change  :  float64
revenues  :  float64
revenue_change  :  float64
profits  :  float64
profit_change  :  float64
assets  :  float64
market_value  :  float64
rank_is_changed  :  float64


## Rank Analysis : 

In [17]:
df['rank_is_changed'] = df['rank_change'].apply(lambda x : False if x==0.0 else True)
rank_change_df = df[df['rank_is_changed']== False]
rank_change_df.reset_index(drop=True, inplace=True)
rank_change_df.head()

rank            company  no_of_employees  rank_change  revenues  \
0     1            Walmart        2200000.0          0.0  523964.0   
1     9               AT&T         247800.0          0.0  181193.0   
2    10  AmerisourceBergen          21500.0          0.0  179589.1   
3    12         Ford Motor         190000.0          0.0  155900.0   
4    14   Costco Wholesale         201500.0          0.0  152703.0   

   revenue_change  profits  profit_change    assets  market_value  \
0            0.02  14881.0           1.23  236495.0      321803.3   
1            0.06  13903.0          -0.28  551669.0      209387.5   
2            0.07    855.4          -0.48   39172.0       18221.4   
3           -0.03     47.0          -0.99  258537.0       19150.6   
4            0.08   3659.0           0.17   45400.0      125907.7   

   rank_is_changed  
0            False  
1            False  
2            False  
3            False  
4            False

In [18]:
rank_increased_companies = df[df['rank_change']>0.0]
rank_increased_companies.reset_index(drop=True, inplace=True)
rank_increased_companies.head()

rank         company  no_of_employees  rank_change  revenues  \
0     2      Amazon.com         798000.0          3.0  280522.0   
1     5      CVS Health         290000.0          3.0  256776.0   
2    11        Alphabet         118899.0          4.0  161857.0   
3    13           Cigna          73700.0         52.0  153566.0   
4    17  JPMorgan Chase         256981.0          1.0  142422.0   

   revenue_change  profits  profit_change     assets  market_value  \
0            0.21  11588.0           0.15   225248.0      970680.1   
1            0.32   6634.0           0.00   222449.0       77375.8   
2            0.18  34343.0           0.12   275909.0      798905.3   
3            2.16   5104.0           0.94   155774.0       65896.7   
4            0.08  36431.0           0.12  2687379.0      276750.1   

   rank_is_changed  
0             True  
1             True  
2             True  
3             True  
4             True

In [19]:
rank_decreased_companies = df[df['rank_change']<0.0]
rank_decreased_companies.reset_index(drop=True, inplace=True)
rank_decreased_companies.head()

rank             company  no_of_employees  rank_change  revenues  \
0     3         Exxon Mobil          74900.0         -1.0  264938.0   
1     4               Apple         137000.0         -1.0  260174.0   
2     6  Berkshire Hathaway         391500.0         -2.0  254616.0   
3     7  UnitedHealth Group         325000.0         -1.0  242155.0   
4     8            McKesson          70000.0         -1.0  214319.0   

   revenue_change  profits  profit_change    assets  market_value  \
0           -0.09  14340.0          -0.31  362597.0      160696.3   
1           -0.02  55256.0          -0.07  338516.0     1112640.8   
2            0.03  81417.0          19.25  817729.0      442897.1   
3            0.07  13839.0           0.16  173889.0      236555.2   
4            0.03     34.0          -0.49   59672.0       21845.0   

   rank_is_changed  
0             True  
1             True  
2             True  
3             True  
4             True

## Top 10 Companies of the Rank Analysis Table 

In [20]:
no_change_list = rank_change_df['company'].tolist()
inc_list = rank_increased_companies['company'].to_list()
dec_list = rank_decreased_companies['company'].to_list()
top10_no_change = no_change_list[:10]
top10_inc = inc_list[:10]
top10_dec = dec_list[:10]
rank_data = {'Rank_Retained' : top10_no_change,
           'Rank_Increased' : top10_inc,
           'Rank_Decreased' : top10_dec
           }
rank_df = pd.DataFrame(rank_data)
rank_df[:10]

Rank_Retained      Rank_Increased            Rank_Decreased
0               Walmart          Amazon.com               Exxon Mobil
1                  AT&T          CVS Health                     Apple
2     AmerisourceBergen            Alphabet        Berkshire Hathaway
3            Ford Motor               Cigna        UnitedHealth Group
4      Costco Wholesale      JPMorgan Chase                  McKesson
5       Cardinal Health           Microsoft                   Chevron
6       Bank of America  Marathon Petroleum            General Motors
7     Dell Technologies          Home Depot  Walgreens Boots Alliance
8  State Farm Insurance             Comcast    Verizon Communications
9                   IBM              Anthem                    Kroger

In [21]:
rank_df.to_csv('Rank Analysis.csv')

In [22]:
rank_analysis = [rank_change_df, rank_increased_companies, rank_decreased_companies]
for i in rank_analysis:
  print(len(i.index))

44
236
220


### No of companies which retained their Rank : 44

### No of companies which improved their Rank : 236

### No of companies whose Rank dropped : 220

## Profit Analysis:

In [23]:
profit_df = df.sort_values(by = ['profits'], ascending = False, ignore_index=True)
profit_df[['rank','company', 'profits']][:10]

rank                 company  profits
0     6      Berkshire Hathaway  81417.0
1     4                   Apple  55256.0
2    21               Microsoft  39240.0
3    17          JPMorgan Chase  36431.0
4    11                Alphabet  34343.0
5    25         Bank of America  27430.0
6    45                   Intel  21048.0
7    30             Wells Fargo  19549.0
8    31               Citigroup  19401.0
9    20  Verizon Communications  19265.0

In [24]:
profit_change_df = df.sort_values(by = ['profit_change'], ascending = False, ignore_index=True)
profit_change_df[['rank','company','profit_change']][:10]

rank                              company  profit_change
0    63                        Cisco Systems         104.65
1   319                              Calpine          76.00
2   351                            Alleghany          20.70
3     6                   Berkshire Hathaway          19.25
4   324                           NRG Energy          15.56
5   478                               NetApp          14.38
6   447                            FM Global          14.36
7    89  Massachusetts Mutual Life Insurance           8.30
8   328                              Newmont           7.23
9   327                           Altice USA           6.38

In [25]:
revenue_change_df = df.sort_values(by = ['revenue_change'], ascending = False, ignore_index=True)
revenue_change_df[['rank','company','revenue_change']][:10]

rank                 company  revenue_change
0    13                   Cigna            2.16
1   390               Andersons            1.68
2   133    United Natural Foods            1.30
3   250   L3Harris Technologies            1.08
4   111               ViacomCBS            0.92
5   388  Westinghouse Air Brake            0.88
6   381       CommScope Holding            0.83
7   349                     KKR            0.75
8   311                  Fiserv            0.75
9   288        Keurig Dr Pepper            0.49

In [26]:
market_value_df = df.sort_values(by = ['market_value'], ascending = False, ignore_index=True)
market_value_df[['rank','company','market_value']][:10]

rank             company  market_value
0    21           Microsoft     1199549.7
1     4               Apple     1112640.8
2     2          Amazon.com      970680.1
3    11            Alphabet      798905.3
4    46            Facebook      475455.2
5     6  Berkshire Hathaway      442897.1
6    35   Johnson & Johnson      345704.9
7     1             Walmart      321803.3
8   137                Visa      316199.1
9    17      JPMorgan Chase      276750.1

In [27]:
employees_df = df.sort_values(by = ['no_of_employees'], ascending = False, ignore_index=True)
employees_df[['rank','company','no_of_employees']][:10]

rank                company  no_of_employees
0     1                Walmart        2200000.0
1     2             Amazon.com         798000.0
2   361     Yum China Holdings         450000.0
3    23                 Kroger         435000.0
4    26             Home Depot         415700.0
5     6     Berkshire Hathaway         391500.0
6    47                  FedEx         389500.0
7    38                    IBM         383800.0
8    43  United Parcel Service         377640.0
9    37                 Target         368000.0

In [28]:
employees_df[['rank','company','no_of_employees']][-10:]

rank                company  no_of_employees
490   319                Calpine           2256.0
491   413          DCP Midstream           2250.0
492   100           StoneX Group           2012.0
493   419           Devon Energy           1800.0
494   461                   Hess           1775.0
495   329        Cheniere Energy           1530.0
496   349                    KKR           1384.0
497   483       EnLink Midstream           1355.0
498   457  Brighthouse Financial           1330.0
499   127    NGL Energy Partners           1300.0

## Summary Table :

In [29]:
df2 = df.set_index('company')
df2[:5]

rank  no_of_employees  rank_change  revenues  revenue_change  \
company                                                                     
Walmart         1        2200000.0          0.0  523964.0            0.02   
Amazon.com      2         798000.0          3.0  280522.0            0.21   
Exxon Mobil     3          74900.0         -1.0  264938.0           -0.09   
Apple           4         137000.0         -1.0  260174.0           -0.02   
CVS Health      5         290000.0          3.0  256776.0            0.32   

             profits  profit_change    assets  market_value  rank_is_changed  
company                                                                       
Walmart      14881.0           1.23  236495.0      321803.3            False  
Amazon.com   11588.0           0.15  225248.0      970680.1             True  
Exxon Mobil  14340.0          -0.31  362597.0      160696.3             True  
Apple        55256.0          -0.07  338516.0     1112640.8             True  
CVS Health    6634.0           0.00  222449.0       77375.8             True

In [31]:
summary_list = ['Walmart','Amazon.com','Exxon Mobil','Apple','CVS Health','Berkshire Hathaway',
                'Pfizer','Alphabet','Facebook','Intel','NetApp','Calpine']
df2.loc[summary_list,['rank','rank_change','revenue_change','profit_change','profits']]

rank  rank_change  revenue_change  profit_change  profits
company                                                                      
Walmart                1          0.0            0.02           1.23  14881.0
Amazon.com             2          3.0            0.21           0.15  11588.0
Exxon Mobil            3         -1.0           -0.09          -0.31  14340.0
Apple                  4         -1.0           -0.02          -0.07  55256.0
CVS Health             5          3.0            0.32           0.00   6634.0
Berkshire Hathaway     6         -2.0            0.03          19.25  81417.0
Pfizer                64         -3.0           -0.04           0.46  16273.0
Alphabet              11          4.0            0.18           0.12  34343.0
Facebook              46         11.0            0.27          -0.16  18485.0
Intel                 45         -2.0            0.02           0.00  21048.0
NetApp               478          5.0            0.04          14.38   1169.0
Calpine              319         11.0            0.06          76.00    770.0